File: Payment/Infrastructure/Config/mongo_connection.py
import os
from pymongo import MongoClient
from dotenv import load_dotenv

load_dotenv()

def mongo_connection():
    client = MongoClient(
        host=os.getenv('DB_HOST_MONGODB'),
        port=int(os.getenv('DB_PORT_MONGODB')),
        username=os.getenv('DB_USER_MONGODB'),
        password=os.getenv('DB_PASSWORD_MONGODB'),
        authSource='admin'
    )
    db = client[os.getenv('DB_DATABASE_MONGODB')]
    return db

File: src/Product/Infrastructure/Config/mongoConnection.ts
import mongoose from 'mongoose';
import { config } from 'dotenv';
config();

const mongoUri = `mongodb://${process.env.DB_HOST_MONGODB}:${process.env.DB_PORT_MONGODB}/${process.env.DB_DATABASE_MONGODB}`;

export const mongoConnection = async () => {
  try {
    await mongoose.connect(mongoUri, {
      user: process.env.DB_USER_MONGODB,
      pass: process.env.DB_PASSWORD_MONGODB,
      authSource: 'admin', //authSource si es necesario
    });
    console.log('Conexión exitosa a la base de datos con MongoDB LISTA!');
  } catch (error) {
    console.error('Error al conectar a la base de datos de MongoDB:', error);
  }
};

// import mongoose from 'mongoose';
// import { config } from 'dotenv';

// config();

// const mongoUri = `${process.env.MONGO_URI}`;

// export const mongoConnection = async () => {
//   try {
//     await mongoose.connect(mongoUri);
//     console.log('Conexión exitosa a la base de datos con MongoDB LISTA!');
//   } catch (error) {
//     console.error('Error al conectar a la base de datos de MongoDB:', error);
//   }
// };
